sw/device/silicon_creator/manuf/base/ft_personalize.c

```diff
@@ -572,14 +580,28 @@ static status_t personalize_endorse_certificates(ujson_t *uj) {
   size_t free_room = sizeof(all_certs);
   // Helper structure caching certificate information.
   perso_tlv_cert_block_t block;
-  // UDS cert.
-  TRY(extract_next_cert(&next_cert, &free_room));
-
-  // Now the two CDI certs which were endorsed locally and sent to the host
-  // earlier.
-  size_t cdis[] = {cdi_0_offset, cdi_1_offset};
-  for (size_t i = 0; i < ARRAYSIZE(cdis); i++) {
-    size_t offset = cdis[i];
+
+  size_t cert_offsets[3] = {SIZE_MAX};
+  size_t cert_offsets_count = 3;
+  if (dice_chain_format == kDiceChainFormatX509) {
+    // UDS cert.
+    TRY(extract_next_cert(&next_cert, &free_room));
+
+    // Now the two CDI certs which were endorsed locally and sent to the host
+    // earlier.
+    cert_offsets[0] = cdi_0_offset;
+    cert_offsets[1] = cdi_1_offset;
+    cert_offsets_count = 2;
+  } else if (dice_chain_format == kDiceChainFormatCWT) {
+    // UDS cert will be in the "perso_blob_to_host" as well. So parse it along
+    // with CDIs
+    cert_offsets[0] = 0;
```

Comment:
Originally, the perso code sent the following from the **device** to the **host**:
1. unendorsed UDS (TBS) cert
2. endorsed CDI_0 cert
3. endorsed CDI_1 cert
4. unendorsed certs created in the perso extension

The **host** then sent back to the **device**:
1. endorsed UDS
2. endorsed certs from the perso extension.
Notice, the host does not send back the CDI certs b/c the device already has them in endorsed form.

Since all DICE certs are written to a page, the original code here was trying to simplify the code further below that does the actual writing to flash by preparing a buffer with the exact ordering of the DICE certs to be written to the flash page, i.e. put UDS, CDI_0, and CDI_1 all in order in a single buffer (`all_certs`) so a that we could loop over them below and write them to flash.

For the CWT case, my understanding is we still want to write all DICE CWT certs to the same page (UDS, CDI_0, CDI_1), except in this case, UDS is also not sent back from the **host** to the **device**, since the device already has it in its end format (i.e., it is not endorsed by anything).

That said, I think what you have is technically correct, but wouldn't this be more readable/less error prone if other items ever come before the UDS cert in the perso blob sent to the host? 

```suggestion
    cert_offsets[0] = uds_offset;
```
(obviously you would need to mark the uds_offset above when it is generated